saleor/graphql/discount/mutations/promotion_create.py

```diff
@@ -0,0 +1,223 @@
+        ),
+    )
+    reward_value = PositiveDecimal(
+        description=(
+            "Defines the discount value. Required when catalogue predicate is provided."
+        ),
+    )
+
+    class Meta:
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+
+class PromotionInput(BaseInputObjectType):
+    description = JSON(description="Promotion description.")
+    start_date = graphene.types.datetime.DateTime(
+        description="The start date of the promotion in ISO 8601 format."
+    )
+    end_date = graphene.types.datetime.DateTime(
+        description="The end date of the promotion in ISO 8601 format."
+    )
+
+
+class PromotionCreateInput(PromotionInput):
+    name = graphene.String(description="Promotion name.", required=True)
+    rules = NonNullList(PromotionRuleInput, description="List of promotion rules.")
+
+    class Meta:
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+
+class PromotionCreate(ModelMutation):
+    class Arguments:
+        input = PromotionCreateInput(
+            description="Fields requires to create a promotion.", required=True
+        )
+
+    class Meta:
+        description = "Creates a new promotion." + ADDED_IN_315 + PREVIEW_FEATURE
+        model = models.Promotion
+        object_type = Promotion
+        permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
+        error_type_class = PromotionCreateError
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+    @classmethod
+    def clean_input(
+        cls, info: ResolveInfo, instance: models.Promotion, data: dict, **kwargs
+    ):
+        cleaned_input = super().clean_input(info, instance, data, **kwargs)
+
+        errors: DefaultDict[str, List[ValidationError]] = defaultdict(list)
+        start_date = cleaned_input.get("start_date")
+        end_date = cleaned_input.get("end_date")
+        try:
+            validate_end_is_after_start(start_date, end_date)
+        except ValidationError as error:
+            error.code = PromotionCreateErrorCode.INVALID.value
+            errors["end_date"].append(error)
+
+        if rules := cleaned_input.get("rules"):
+            cleaned_rules, errors = cls.clean_rules(info, rules, errors)
+            cleaned_input["rules"] = cleaned_rules
+
+        if errors:
+            raise ValidationError(errors)
+
+        return cleaned_input
+
+    @classmethod
+    def clean_rules(
+        cls,
+        info: ResolveInfo,
+        rules_data: dict,
+        errors: DefaultDict[str, List[ValidationError]],
+    ):
+        cleaned_rules = []
+        for index, rule_data in enumerate(rules_data):
+            if channel_ids := rule_data.get("channels"):
+                channels = cls.clean_channels(info, channel_ids, index, errors)
+                rule_data["channels"] = channels
+
+            if "catalogue_predicate" not in rule_data:
+                errors["catalogue_predicate"].append(
+                    ValidationError(
+                        "The cataloguePredicate field is required.",
```

Comment:
Do we use camel case in errors?

Reply:
The error is related with the field `catalogue_predicate`, so API will return it as a error assigned to the field `cataloguePredicate`. In that case we could just have an error msg: `The field is required.`

Reply:
This is the message that is returning in the API so IMO it should be camel case and we're doing this in other places.